db/hints/manager.cc

```diff
@@ -286,18 +287,8 @@ sync_point::shard_rps manager::calculate_current_sync_point(std::span<const gms:
 }
 
 future<> manager::wait_for_sync_point(abort_source& as, const sync_point::shard_rps& rps) {
-    using lock_type = std::shared_lock<seastar::shared_mutex>;
     // Prevent the migration to host-ID-based hinted handoff until this function finishes its execution.
-    const auto shared_lock = co_await std::invoke(coroutine::lambda([&] () -> future<lock_type> {
-        co_await _migration_mutex.lock_shared();
-
-        try {
-            co_return lock_type{_migration_mutex, std::adopt_lock_t{}};
-        } catch (...) {
-            _migration_mutex.unlock_shared();
-            throw;
-        }
-   }));
+    const auto shared_lock = co_await get_shared_lock(_migration_mutex);
```

Comment:
Now that we use `get_{shared,unique}_lock` and don't name the type of the lock I think we can also drop the `<shared_mutex>` import, right?

Reply:
Yes, that's a good point